### ISSUE TITLE
AS3 to expose complete control over the options property of SSL profiles.

### ISSUE THREAD

**Issue by trinaths** [F5Networks/f5-appsvcs-extension#233]
While creating SSL profiles using custom certificates with AS3, there is no option to update "Options List" with "Enabled Options" along with ciphers. 

. . .
**options { dont-insert-empty-fragments single-dh-use no-sslv3 no-dtls no-tlsv1 no-tlsv1.1 }**
. . .

in TMSH, 
```

ltm profile client-ssl TLS_SERVER_PROFILE- {
   ciphers none
   cipher-group /Common/MY_CIPHER
   options { dont-insert-empty-fragments single-dh-use no-sslv3 no-dtls no-tlsv1 no-tlsv1.1 }
}

```
we need an option to enable "options list" from AS3.

More info on SSL profile Options list - https://techdocs.f5.com/content/kb/en-us/products/big-ip_ltm/manuals/product/bigip-ssl-administration-11-6-0/5.html

**Comment by dstokesf5** [comment]
This is in our internal product backlog as AUTOTOOL-1234. Are the options you specified the only ones you need to configure at this time?

**Comment by trinaths** [comment]
> This is in our internal product backlog as AUTOTOOL-1234. Are the options you specified the only ones you need to configure at this time?

Please support the options available with BIG-IP (listed below). 
This will enable user with update options as required.

- Don't insert empty fragments.
- No TLSv1.3
- No DTLS
- No SSL
- No session resumption on renegotiation.
- No TLSv1.1
- No TLSv1.2
- No TLS
- Single DH use
- TLS rollback bug workaround.
- No SSLv3
- No TLSv1

**Comment by notandy** [comment]
We also urgently need this feature or at least parent profile definition.

**Comment by m-kratochvil** [comment]
Please, could this be worked on with urgency? This is very important piece of configuration, especially these days where modification of TLS protocol support is a global topic.
Thank you!

**Comment by mdditt2000** [comment]
@notandy @m-kratochvil feature has been coded. We have an internal dev image. Would you like to test this feature, provide some feedback etc until we get a release candidate in April

**Comment by notandy** [comment]
> @notandy @m-kratochvil feature has been coded. We have an internal dev image. Would you like to test this feature, provide some feedback etc until we get a release candidate in April

Sounds good! Sure we'll be happy to test it.

**Comment by m-kratochvil** [comment]
> @notandy @m-kratochvil feature has been coded. We have an internal dev image. Would you like to test this feature, provide some feedback etc until we get a release candidate in April

Can you provide us the dev image please?

**Comment by dstokesf5** [comment]
@m-kratochvil AS3 should have an official release containing this feature available within the next couple of days.

**Comment by dstokesf5** [comment]
@m-kratochvil AS3 will have an official release available within the next couple of days that includes this feature.

**Comment by m-kratochvil** [comment]
> @m-kratochvil AS3 will have an official release available within the next couple of days that includes this feature.

Very good, thank you for the confirmation, we will wait for the official release then.

**Comment by wncocz** [comment]
Is there support for "No SSLv3" or "No SSL" in v3.19?  I don't see it documented, and it is ridiculous that virtuals we configure with no TLS1.0 or TLS1.1 support still allow SSLv3 with no apparent way to control that in AS3.  We still cannot configure "TLSv1.2-only" virtuals on BIG-IP 13.1.3 with AS3, yet this is supposed to be our standard procedure.

**Comment by dstokesf5** [comment]
@wncocz We implemented a list of settings that was supplied out of band for this issue for a customer's use case. If there are some more options you need exposed, please file a new ticket listing the options you need.

**Comment by wncocz** [comment]
The release was billed above as "includes this feature."  If I knew that "this feature" meant something that was supplied out of band as opposed to what is documented here, I would have taken action sooner.  Please provide clearer communication in the future.

I will open another support case when I get time to write it up.

**Comment by dstokesf5** [comment]
This has been resolved as part of the 3.19.0 release.
